[Bug]: Typescript Client Package should use the Release tag version

### Describe the bug

I released MemMachine version 0.3.0. The Python packages successfully published to pypi.org using the new `v0.3.0` tag. The Typescript client package failed as it uses a hard-coded version.

Q) Should we keep the Typescript client package version coupled with the MemMachine version or separate? It had no changes.

The GitHub Action logs report 

```
Run npm publish --access public
npm notice
npm notice 📦  @memmachine/client@0.2.5
npm notice Tarball Contents
npm notice 4.4kB README.md
npm notice 19.5kB dist/index.d.mts
npm notice 19.5kB dist/index.d.ts
npm notice 16.2kB dist/index.js
npm notice 29.4kB dist/index.js.map
npm notice 14.3kB dist/index.mjs
npm notice 28.9kB dist/index.mjs.map
npm notice 2.1kB package.json
npm notice Tarball Details
npm notice name: @memmachine/client
npm notice version: 0.2.5
npm notice filename: memmachine-client-0.2.5.tgz
npm notice package size: 16.5 kB
npm notice unpacked size: 134.4 kB
npm notice shasum: 9b279b9e2da2cb5d4b65ed57c67fdd28407cee5c
npm notice integrity: sha512-8uyD4zviqvF+T[...]Rbg9PCWn2MZtw==
npm notice total files: 8
npm notice
npm notice Publishing to https://registry.npmjs.org/ with tag latest and public access
npm error code E403
npm error 403 403 Forbidden - PUT https://registry.npmjs.org/@memmachine%2fclient - You cannot publish over the previously published versions: 0.2.5.
npm error 403 In most cases, you or one of your dependencies are requesting
npm error 403 a package version that is forbidden by your security policy, or
npm error 403 on a server you do not have access to.
npm error A complete log of this run can be found in: /home/runner/.npm/_logs/2026-03-04T23_14_39_694Z-debug-0.log
Error: Process completed with exit code 1.
```

This is the workflow file

https://github.com/MemMachine/MemMachine/blob/main/.github/workflows/ts-client-publish.yml

### Steps to reproduce

Perform a new MemMachine release or run the workflow file manually.
The workflow fails when the version isn't bumped and there is already a released version upstream in NPMJS.

### Expected behavior

The action should complete successfully

### Environment

GitHub

### Additional context

We need to manually increment the package version before publishing, or see if it's possible to automatete this using the Workflow and release tag. Update the version field in your package.json file to a new, unused version (for example, 0.3.0).

--- a/.github/workflows/ts-client-publish.yml
+++ b/.github/workflows/ts-client-publish.yml
@@ -5,6 +5,11 @@ name: Publish TypeScript Client Package to npm
 on:
   release:
     types: [published]
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to publish (e.g. 0.3.0, without v prefix)'
+        required: true
 
 permissions:
   contents: read
@@ -24,6 +29,17 @@ jobs:
         with:
           node-version: '20.19.x'
           registry-url: 'https://registry.npmjs.org/'
+
+      - name: Set version from release tag
+        env:
+          INPUT_VERSION: ${{ inputs.version }}
+        run: |
+          if [ "$GITHUB_EVENT_NAME" = "workflow_dispatch" ]; then
+            VERSION="$INPUT_VERSION"
+          else
+            VERSION="${GITHUB_REF_NAME#v}"
+          fi
+          npm version "$VERSION" --no-git-tag-version
 
       - name: Install dependencies
         run: npm install

--- a/packages/ts-client/package.json
+++ b/packages/ts-client/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@memmachine/client",
-  "version": "0.2.5",
+  "version": "0.0.0-development",
   "description": "TypeScript client for MemMachine RESTful APIs",
   "homepage": "https://memmachine.ai",
   "repository": {

